src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -279,23 +275,30 @@ object RuleUtils {
             baseOutput,
             _,
             _) =>
-        val curFiles = location.allFiles
-          .map(f => FileInfo(f, index.fileIdTracker.addFile(f), asFullPath = true))
-
         val (filesDeleted, filesAppended) =
-          if (HyperspaceConf.hybridScanDeleteEnabled(spark) && index.hasLineageColumn) {
-            val (exist, nonExist) = curFiles.partition(index.sourceFileInfoSet.contains)
-            val filesAppended = nonExist.map(f => new Path(f.name))
-            if (exist.length < index.sourceFileInfoSet.size) {
-              (index.sourceFileInfoSet -- exist, filesAppended)
+          if (!HyperspaceConf.hybridScanEnabled(spark) && index.hasSourceUpdate) {
+            // This index was refreshed in quick mode and validated with the latest signature.
```

Comment:
Same here. I don't think we need to mention "quick" mode. You can say that the index data is not up to date with source data if index.hasSourceUpdate is true.

Reply:
Good point, updated the comments :)